### DISCUSSION TITLE
Observable Properties not generating in azure pipelines

### DISCUSSION THREAD

**Discussion by Jalroka** [CommunityToolkit/dotnet#567] [Q&A] [answered]
I'm trying to use [ObservableProperty] to reduce the code in my files.  I have no issues running the application I'm working on using Visual Studio.  But when I try to build the application in Azure pipelines I get errors such as this.  
/Users/runner/work/1/s/maskedName.***.Mobile.Forms/AppModels/Catalog/FiltersModel.cs(45,21): error CS0117: 'CheckBoxItem' does not contain a definition for 'Text' [/Users/runner/work/1/s/maskedName..***.Mobile.Forms/maskedName..***.Mobile.Forms.csproj]

For a code snippet 
`public partial class CheckBoxItem : ObservableObject
    {
        [ObservableProperty]
        string text;

        [ObservableProperty]
        string subText;

        [ObservableProperty]
        bool isChecked;

        [ObservableProperty]
        string grouping;
    }`

Is there something special that I need to add to my azure pipeline yml to get the source code generation to work?

**Comment by Sergio0694** [comment]
Just to double check, are you using packages.config by any chance? But no, you don't need anything special to get generators to run in a pipeline, a normal dotnet or MSBuild invocation will work (it's just a normal build).

**Reply by Jalroka** [reply]
No I'm not using packages.config.

**Reply by Sergio0694** [reply]
Is your pipeline using an older version of Roslyn perhaps? Check to see if there are any warnings. The MVVM Toolkit generators need at least Roslyn 4.0 to work (ie. VS2022 or the .NET 6 SDK).

**Reply by Jalroka** [reply]
I've set the pipeline to use .net 7, but I haven't found an option anywhere that lets me specify the Roslyn version.  Any tips on how I can actually do that?

**Reply by Sergio0694** [reply]
If you have the .NET 7 SDK and you're using that to build then that will already have Roslyn 4.4. Not sure what's going on here, your best bet would probably be to get a binary build log form your CI (build with `/bl`) and then inspect it to see what's going on.

**Reply by Jalroka** [reply]
Thank you for your answers.  I'm posting the relevant bits I'm trying with my current yml.  If you see anything that seems odd please let me know.  
`name: "20$(Date:yy.MM.dd)$(Rev:rr)"
jobs:

- job: Android
  condition: eq(variables.BuildAndroid, true)
  pool:
    vmImage: 'macOS-11'
  
  variables:
    buildConfiguration: 'Release'
    outputDirectory: '$(build.binariesDirectory)/$(buildConfiguration)'
    DotNet.Tools: $(Agent.ToolsDirectory)/dotnet
    DotNet.Version: 7.0.x

  steps:
  - script: sudo $AGENT_HOMEDIRECTORY/scripts/select-xamarin-sdk.sh 6_12_16
    displayName: 'Select Xamarin SDK version'
    enabled: false
    
  # Install the .NET 7 SDK
  - task: UseDotNet@2
    displayName: Install the .NET 7 SDK
    inputs:
      version: $(DotNet.Version)
      performMultiLevelLookup: true
      

  - task: NuGetAuthenticate@0
    inputs:
      nuGetServiceConnections: 'masked'
          
  - task: NuGetToolInstaller@1

  - task: NuGetCommand@2
    inputs:
      restoreSolution: '**/*.sln'
      feedsToUse: 'Config'
      nugetConfigPath: 'NuGet.config'
      externalFeedCredentials: 'masked'
      
  - script: chmod +x $(build.sourcesDirectory)/masked.Mobile.Droid/pre-build.sh
    displayName: 'Set Execute on pre-build'
    
  - script: bash $(build.sourcesDirectory)/masked.Mobile.Droid/pre-build.sh $(build.sourceBranchName) '$(build.sourcesDirectory)' $(build.buildNumber)
    displayName: 'Execute pre-build'

  - task: XamarinAndroid@1
    inputs:
      projectFile: '**/masked.Mobile.Droid/*.csproj'
      outputDirectory: '$(outputDirectory)'
      configuration: '$(buildConfiguration)'
      msbuildLocationOption: 'location'      
      msbuildFile: '/Library/Frameworks/Mono.framework/Versions/Current/Commands/msbuild'
      jdkOption: 'JDKVersion'
      jdkVersionOption: '1.11'`

**Reply by Jalroka** [reply]
Finally got this working it required adding a couple of nuget packages that aren't required when running in visual studio


Microsoft.CodeAnalysis.Compilers (this includes all roslyn references for C#)

Microsoft.Net.Compilers.Toolset (this includes a specific version os MSBuild and uses that version to build always to avoid issues with different versioning on the pipeline)

**Comment by Jalroka** [comment, marked as answer]
Microsoft.CodeAnalysis.Compilers (this includes all roslyn references for C#)

Microsoft.Net.Compilers.Toolset (this includes a specific version os MSBuild and uses that version to build always to avoid issues with different versioning on the pipeline)
